fip-0040.md

```diff
@@ -356,29 +345,28 @@ Returns all permissions for supplied object.
 |Group|Parameter|Format|Definition|
 |---|---|---|---|
 ||permissions|JSON Array|Array of domain permissions|
-||permissions|JSON Array|Array of domain permissions|
-|permissions|grantee_account|Yes|FIO Chain account|Account receiving permission.|
-|permissions|object_type|Yes|String|Custom name of permission type.|
-|permissions|object_name|Yes|String|Name associated with object.|
-|permissions|aux_info|Yes|String|Aux info.|
-|permissions|actor|Yes|12 character string|Valid actor of signer|
+|permissions|grantee_account|string|FIO Chain account. Account receiving permission.|
+|permissions|permission_name|String|Custom name of permission.|
+|permissions|object_name|String|Name associated with object.|
+|permissions|permission_info|String|optional for a permission so this may be blank, detailed permission settings json.|
+|permissions|actor|12 character string|Valid actor of signer|
 ||more|Int|Number of remaining results|
 ##### Example
 ```
 {
 	"permissions": [
 		{
 			"grantee_account": "deshputyz",
-			"object_type": "regondomain",
+			"permission_name": "register_address_on_domain",
+			"permission_info":"",
 			"object_name": "alice",
-			"aux_info": "",
 			"actor": "aftyershcu22"
 		},
 		{
-			"grantee_account": "asdftredg",
-			"object_type": "regondomain",
-			"object_name": "alice",
-			"aux_info": "",
+			"grantee_account": "deshputyz",
+			"permission_name": "register_address_on_domain",
+			"permission_info":"",
+			"object_name": "bob",
```

Comment:
Since the object_name is passed in, and in this case it is "Alice" then the getter should not return anything with "object_name" : "bob".

But, since objects are not checked, it could return the same object from another actor (assuming two different actors mapped the same object, which is possible).

Reply:
Eric and I are taking this conversation offline to determine the actionable items that might result from this comment. I recommend merging this new version of the FIP and we will revise as necessary during the detailed design since this version is a big step towards finalizing the FIP